src/Channel.js

```diff
@@ -67,27 +72,40 @@ class Channel {
       throw new NoPartyError();
     }
 
+    if (this.audience instanceof PrivateAudience && !targets.length) {
+      throw new NoRecipientError();
+    }
+
     // Allow audience to change message e.g., strip target name.
     message = this.audience.alterMessage(message);
 
-    // Private channels also send the target player to the formatter
-    if (this.audience instanceof PrivateAudience) {
-      if (!targets.length) {
-        throw new NoRecipientError();
+    // Send messages with Broadcast unless the channel is eventOnly.
+    if (!this.eventOnly) {
+      // Private channels also send the target player to the formatter
+      if (this.audience instanceof PrivateAudience) {
+        Broadcast.sayAt(sender, this.formatter.sender(sender, targets[0], message, this.colorify.bind(this)));
+      } else {
+        Broadcast.sayAt(sender, this.formatter.sender(sender, null, message, this.colorify.bind(this)));
       }
-      Broadcast.sayAt(sender, this.formatter.sender(sender, targets[0], message, this.colorify.bind(this)));
-    } else {
-      Broadcast.sayAt(sender, this.formatter.sender(sender, null, message, this.colorify.bind(this)));
+  
+      // send to audience targets
+      Broadcast.sayAtFormatted(this.audience, message, (target, message) => {
+        return this.formatter.target(sender, target, message, this.colorify.bind(this));
+      });
     }
 
-    // send to audience targets
-    Broadcast.sayAtFormatted(this.audience, message, (target, message) => {
-      return this.formatter.target(sender, target, message, this.colorify.bind(this));
-    });
-
     // strip color tags
     const rawMessage = message.replace(/\<\/?\w+?\>/gm, '');
 
+    // Emit channel events
+
+    /**
+     * @event GameEntity#channelSend
+     * @param {Channel} channel
+     * @param {string} rawMessage
+     */
+    sender.emit('channelSend', this, rawMessage);
```

Comment:
This PR has been open for a bit and I recently merged it into my local fork and tried using it. One thing that stood out is that it may be helpful to actually emit this on the audience if possible... Mainly, I would like to handle sending to different audience members in different ways depending on whether they use telnet or webclient.

Reply:
Hmm, on second thought, it is still possible to do with the way this is set up. The one thing it may make a bit harder is having NPCs use the channels but that can be done with a behavior.